src/rust/aggregator/src/data/markets.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::anyhow;
+use chrono::{DateTime, Duration, Utc};
+use sqlx::PgPool;
+
+use super::{Data, DataAggregationError, DataAggregationResult};
+
+pub struct MarketsRegisteredPerDay {
+    pool: PgPool,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+impl MarketsRegisteredPerDay {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Data for MarketsRegisteredPerDay {
+    fn ready(&self) -> bool {
+        self.last_indexed_timestamp.is_none()
+            || self.last_indexed_timestamp.unwrap() + Duration::days(1) < Utc::now()
+    }
+
+    async fn process_and_save_historical_data(&mut self) -> DataAggregationResult {
+        sqlx::query!(
+            r#"
+                INSERT INTO aggregator.markets_registered_per_day (markets, date)
+                SELECT COUNT(*), time::date
+                FROM market_registration_events
+                WHERE time::date NOT IN (
+                    SELECT date FROM aggregator.markets_registered_per_day
```

Comment:
We should probably not be using timestamp this way. I realize it's unlikely but if there are two events with the same timestamp then we'll miss one. Market registration events have a transaction version number and event index and this comprises a unique identifier. Let's use that.